src/brsTypes/components/Regex.ts

```diff
@@ -0,0 +1,153 @@
+                let arr = [];
+                if (result !== null) {
+                    for (let i = 0; i < result.length; i++) {
+                        let item = result[i] ? new BrsString(result[i]) : BrsInvalid.Instance;
+                        arr.push(item);
+                    }
+                }
+
+                return new BrsArray(arr);
+            }
+        }
+    );
+
+    private replace = new Callable(
+        "replace",
+        {
+            signature: {
+                args: [
+                    new StdlibArgument("str", ValueKind.String),
+                    new StdlibArgument("str", ValueKind.String)
+                ],
+                returns: ValueKind.String
+            },
+            impl: (interpreter: Interpreter, str: BrsString, replacement: BrsString) => {
+                let replacementPattern = this.parseReplacementPattern(replacement.value);
+                const newStr = this.jsRegex[Symbol.replace](str.value, replacementPattern);
+                return new BrsString(newStr);
+            }
+        }
+    );
+
+    private replaceAll = new Callable(
+        "replaceall",
+        {
+            signature: {
+                args: [
+                    new StdlibArgument("str", ValueKind.String),
+                    new StdlibArgument("str", ValueKind.String)
+                ],
+                returns: ValueKind.String
+            },
+            impl: (interpreter: Interpreter, str: BrsString, replacement: BrsString) => {
+                const source = this.jsRegex.source;
+                const flags = this.jsRegex.flags + "g";
+                this.jsRegex = new RegExp(source, flags);
+                const newStr = this.jsRegex[Symbol.replace](str.value, replacement.value);
+
+                return new BrsString(newStr);
+            }
+        }
+    );
+
+    private split = new Callable(
+        "split",
+        {
+            signature: {
+                args: [
+                    new StdlibArgument("str", ValueKind.String)
+                ],
+                returns: ValueKind.Object
+            },
+            impl: (interpreter: Interpreter, str: BrsString) => {
+                let items = this.jsRegex[Symbol.split](str.value);
+                let brsItems = items.map( item => new BrsString(item));
+                return new BrsArray(brsItems);
+            }
+        }
+    );
+
+    private matchAll = new Callable(
+        "matchall",
+        {
+            signature: {
+                args: [
+                    new StdlibArgument("str", ValueKind.String)
+                ],
+                returns: ValueKind.Object
+            },
+            impl: (interpreter: Interpreter, str: BrsString) => {
+                const source = this.jsRegex.source;
+                const flags = this.jsRegex.flags + "g";
+                this.jsRegex = new RegExp(source, flags);
+                let arr = [];
+                let matches:any;
```

Comment:
Looks like we're using `matches` as an array, correct?  Looks like its type might be `: string[] | null` – would you mind updating this?